core/controllers/base_test.py

```diff
@@ -712,6 +770,34 @@ def test_controller_class_names(self):
         self.assertGreater(num_handlers_checked, 150)
 
 
+class HasSeenTutorialTests(test_utils.GenericTestBase):
+
+    class MockHandler(base.BaseHandler):
```

Comment:
Refrain from using mocks unless absolutely necessary. Otherwise it's possible for your tests to pass but the production code to be failing.

Reply:
This mock class is required to check if the values of `self.has_seen_editor_tutorial` and `self.has_seen_translation_tutorial` changes properly or not and that is only possible by creating a mock class that inherits from the BaseHandler

Reply:
@seanlip @brianrodri I made `has_seen_translation_tutorial` and `has_seen_editor_tutorial` accessible to only 'ExplorationHandler' as they are only used here.